src/main/scala/calc/Typer.scala

```diff
@@ -0,0 +1,33 @@
+package calc
+
+object Typer {
+
+  type TypeEnv = Map[Ident, Type]
```

Comment:
Relying on equality of trees is risky. Typically, an environment is a map form `String` to `Type`, where the string is the `name` of an `Ident`, not an `Ident` itself.